src/tir/transforms/compact_buffer_region.cc

```diff
@@ -469,7 +507,8 @@ class BufferCompactor : public StmtExprMutator {
     // Step 0. Check there is no Init part.
     ICHECK(!op->init.defined());
     // Step 1. Reallocate and rewrite alloc_buffers, also update BufferAllocInfo.
-    Array<Buffer> alloc_buffers = RewriteAllocBuffer(op->alloc_buffers);
+    Array<Buffer> alloc_buffers = op->alloc_buffers.Map(
```

Comment:
Nit: Personal preference, but I think lambda expressions tend to be more readable than the equivalent `std::bind` expression.  Here, this could be rewritten as `op->alloc_buffers.Map([this](const Buffer& buf) { return RewriteAllocBuffer(buf); });`